### PR TITLE
Add Flag to Disable Exec

### PR DESCRIPTION
**Pull request by dr-bonez**
This PR addresses #1088 through the following:

Frontend:
- Added settings as global state
- Removed all ui elements pertaining to command execution if settings.execEnabled is false

Backend:
- added a config flag: `execEnabled` default `true`
- disabled command runner and hooks if execEnabled is false

I have never used Vue.js before today, so let me know if I did anything un-idiomatic and I'll change it
I have also only dabbled in golang before so same caviat there

### CODE REVIEW

File: cmd/config.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ func addConfigFlags(flags *pflag.FlagSet) {
 	addServerFlags(flags)
 	addUserFlags(flags)
 	flags.BoolP("signup", "s", false, "allow users to signup")
+	flags.BoolP("execEnabled", "", true, "enable the Command Runner feature")
```
Comment (o1egl) [inline cmd/config.go:34]: Let's call it `disable-exec`

File: frontend/src/main.js [hunk 2]
```diff
@@ -15,6 +16,7 @@ async function start () {
   } else {
     await login('', '', '')
   }
+  store.commit('setSettings', await settings.get())
```
Comment (ramiresviana) [inline frontend/src/main.js:19]: Getting settings requires authentication, causing hangs on login screen.

**Comment by ramiresviana** [comment]
Using settings with states is a nice approach, but this implementation covers only the execEnable param, leaving the remaining settings as constants. Having multiple pieces of code for managing settings adds complexity to the frontend, requiring contributors to know both implementations, that can be bad for maintainability.

Since this is not the scope of this PR, i recommend using only constants for the settings.

**Comment by dr-bonez** [comment]
Can you point me to an example of the constants?

**Comment by ramiresviana** [comment]
@dr-bonez you can implement params and settings like this commit: https://github.com/filebrowser/filebrowser/commit/aa78e3ab1fcae6f618e811ba4e315a7a209f9df2

**Review by o1egl — COMMENTED** [review]
Totally agree with @ramiresviana

**Comment by dr-bonez** [comment]
closing in favor of #1090

**Pull request closed without merging**